### PR TITLE
feat(jkube-kit): Initial Support for CronJob (#1954)

### LINKED ISSUE #1954: Support for CronJob controller resources
Hi like job in jkube, we can create directly job, such thing is not available for cron job. can you please create for cronjob like job. there should be jkube.enricher.jkube-controller.type for cronjob too

### PR DESCRIPTION
**Pull request by anurag-rajawat**
## Description

Fixes #1954 

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [x] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] I have read the [contributing guidelines](https://www.eclipse.org/jkube/contributing)
 - [x] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [x] My code follows the style guidelines of this project
 - [ ] I Keep It Small and Simple: The smaller the PR is, the easier it is to review and have it merged
 - [x] I use [conventional commits](https://www.conventionalcommits.org/) in my commit messages
 - [x] I have performed a self-review of my code
 - [x] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [ ] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [x] I have added tests that prove my fix is effective or that my feature works
 - [x] New and existing unit tests pass locally with my changes
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

File: jkube-kit/enricher/api/src/main/java/org/eclipse/jkube/kit/enricher/handler/PodTemplateHandler.java [hunk 1]
```diff
@@ -75,6 +79,16 @@ private List<Volume> getVolumes(ResourceConfig config) {
...
+          .withSpec(createJobSpec(config, restartPolicy, images))
+          .build();
+    }
 
-
+    private JobSpec createJobSpec(ResourceConfig config, String restartPolicy, List<ImageConfiguration> images) {
+      return new JobSpecBuilder()
+          .withTemplate(getPodTemplate(config, restartPolicy, images))
+          .build();
+    }
```
Comment (rohanKanojia) [inline jkube-kit/enricher/api/src/main/java/org/eclipse/jkube/kit/enricher/handler/PodTemplateHandler.java:93]: Umm, this doesn't look like the right place for these methods. Do you think they can be shifted somewhere else?

File: jkube-kit/enricher/api/src/test/java/org/eclipse/jkube/kit/enricher/handler/CronJobHandlerTest.java [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+  @Test
+  void overrideReplicas() {
+    // Given
+    KubernetesListBuilder builder = new KubernetesListBuilder().addToItems(new CronJob());
+    // When
+    cronJobHandler.overrideReplicas(builder, 1337);
+    // Then
+    assertThat(builder.buildItems())
+        .singleElement()
+        .isEqualTo(new CronJob());
```
Comment (rohanKanojia) [inline jkube-kit/enricher/api/src/test/java/org/eclipse/jkube/kit/enricher/handler/CronJobHandlerTest.java:166]: This test doesn't seem to be testing anything. It's better if we remove this test.

File: jkube-kit/enricher/api/src/main/java/org/eclipse/jkube/kit/enricher/handler/PodTemplateHandler.java (hunk 1 above)
Comment (anurag-rajawat) [reply jkube-kit/enricher/api/src/main/java/org/eclipse/jkube/kit/enricher/handler/PodTemplateHandler.java:93]: Would the `CronJobHandle` class be the right place for these methods?

Comment (rohanKanojia) [reply jkube-kit/enricher/api/src/main/java/org/eclipse/jkube/kit/enricher/handler/PodTemplateHandler.java:93]: If it's used only by CronJobHandler, then it makes sense to move there there.

File: jkube-kit/doc/src/main/asciidoc/inc/enricher/controller/_jkube_controller.adoc [hunk 3]
```diff
@@ -38,7 +38,7 @@ Image pull policy to use for the container. One of: _IfNotPresent_, _Always_.
 
 | *type*
 | Type of Controller to create. One of: _ReplicationController_, _ReplicaSet_, _Deployment_, _DeploymentConfig_,
-  _StatefulSet_, _DaemonSet_, _Job_.
+  _StatefulSet_, _DaemonSet_, _Job_, _CronJob_.
```
Comment (rohanKanojia) [inline jkube-kit/doc/src/main/asciidoc/inc/enricher/controller/_jkube_controller.adoc:41]: I think we also need to add a row  for `schedule` field

File: jkube-kit/doc/src/main/asciidoc/inc/resource-generation/_controller_resource_generation.adoc [hunk 4]
```diff
@@ -11,6 +11,7 @@ In JKube terminology, a Controller resource is a Kubernetes resource which manag
 * https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/[StatefulSet]
 * https://kubernetes.io/docs/concepts/workloads/controllers/daemonset/[DaemonSet]
 * https://kubernetes.io/docs/concepts/workloads/controllers/job/[Job]
+* https://kubernetes.io/docs/concepts/workloads/controllers/cron-jobs/[CronJob]
```
Comment (rohanKanojia) [inline jkube-kit/doc/src/main/asciidoc/inc/resource-generation/_controller_resource_generation.adoc:14]: I think we need to add new field `schedule` to `.controller fields for configuring generated controllers` table

**Comment by rohanKanojia** [comment]
We would also need to add an integration test for CronJob like the rest of the controllers here:

https://github.com/eclipse/jkube/blob/2b1891f555019900edf3544d6a41aabf5fc384dc/gradle-plugin/it/src/test/java/org/eclipse/jkube/gradle/plugin/tests/KubernetesConfiguredControllerIT.java#L32-L34

This can be done in a follow-up PR.

**Comment by anurag-rajawat** [comment]
@rohanKanojia I'll open a new PR for integration tests on top of this PR.

**Pull request merged**